sdk/state/update.go

```diff
@@ -137,6 +139,9 @@ func (c *Channel) ConfirmPayment(ca CloseAgreement) (closeAgreement CloseAgreeme
 			(!c.initiator && ca.Details.Balance.Amount < c.latestAuthorizedCloseAgreement.Details.Balance.Amount) {
 			return ca, authorized, fmt.Errorf("close agreement is a payment to the proposer")
 		}
+		if c.amountToLocal(ca.Details.Balance.Amount) > c.remoteEscrowAccount.Balance {
+			return ca, authorized, fmt.Errorf("close agreement over commits: %w", ErrUnderfunded)
+		}
 		txClose, err = txClose.Sign(c.networkPassphrase, c.localSigner)
```

Comment:
💭  thoughts on also adding a check when proposing a payment. To check that the amount I'm sending is not greater than what I have available. More of a self-check for accidentally proposing too much.

Bc if I do propose too much, and the other party decides to accept it, then the channel is in trouble

Reply:
Good idea. Let's address that separate to this. I created an issue for it: https://github.com/stellar/experimental-payment-channels/issues/157.